.ci

```diff
@@ -0,0 +1,158 @@
+    LC_ALL = "en_US.UTF-8"
+    PRODUCTION_DEPLOY="false"
+    SLACK_CHANNEL = "#staging-releases"
+  } // environment
+
+  agent {
+    kubernetes {
+      label k8s_label
+      defaultContainer 'jnlp'
+      yaml """
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test
+spec:
+  restartPolicy: Never
+  containers:
+  - name: ruby
+    image: gcr.io/alist-development/ruby:2.7.5
+    imagePullPolicy: Always
+    resources:
+      requests:
+        memory: "1024Mi"
+        cpu: "1"
+      requests:
+        memory: "1024Mi"
+        cpu: "1"
+    command:
+    - "tail"
+    - "-f"
+    - "/dev/null"
+"""
+    } // kubernetes
+  } // agent
+
+  options {
+    timestamps()
+    timeout(time: 10, unit: 'MINUTES')
+    ansiColor('xterm')
+  } // options
+
+  stages {
+    stage('Preparation') {
+      parallel {
+        stage('Slack') {
+          steps {
+            slackPreparation()
+          } // steps
+        } // stage - Slack
+        stage('Build Description') {
+          steps {
+            buildDescription()
+          }
+        }
+        stage('Bundle') {
+          steps {
+            withCredentials([usernamePassword(credentialsId: 'ApartmentList-CI', usernameVariable: 'GIT_USERNAME', passwordVariable: 'GIT_TOKEN')]) {
+              container('ruby') {
+                withEnv([
+                  "GIT_USERNAME=${GIT_USERNAME}",
+                  "GIT_TOKEN=${GIT_TOKEN}"
+                ]) {
+                  sh 'apt-get update'
+                  sh 'gem install bundler:2.2.6 --no-post-install-message --no-document'
+                  sh 'bundle config set --global rubygems.pkg.github.com $GIT_USERNAME:$GIT_TOKEN'
+                  sh 'bundle install -j 12'
+                } // withEnv
+              } //container
+            } // withCredential
+          } //steps
+        } // stage - Bundle
+      } //parallel
+    } //stage - Preparation
+
+    stage('Testing') {
+      when {
+        allOf {
+          changeRequest target: 'main'
+        }
+      }
+
+      parallel {
+        stage('Run RSpec') {
+          steps {
+            container('ruby') {
+              withEnv([
+                "BUNDLE_GITHUB__COM=${BUNDLE_GITHUB__COM}",
+                "BUNDLE_GITHUB__HTTPS=${BUNDLE_GITHUB__HTTPS}"
+              ]) {
+                sh label: 'RSpec', script: 'bundle exec rake'
+              } // withEnv
+            } // container
+          } // steps
+        } // stage
+      } //parallel
+    } //stage
+
+    stage("Publish") {
+      when {
+        allOf {
+          branch "main"
+          not { changeRequest() }
+        }
+      } // when
+      parallel {
+        stage("to Github packages") {
+          steps {
+            publishRubyGemToGHP("ruby")
+          }
+        } // stage - to Github packages
+      } // parallel
+    } // stage - Publish
+  } // stages
+
+  post {
+    success {
+      slackSuccess()
+    } // success
+    failure {
+      slackFailure()
+    } // failure
```

Comment:
we now like this updated way

```
success {
  success('alist-staging')
} // success
aborted {
  aborted('alist-staging')
} // aborted
```

Reply:
and actually this change will require a few more env values:

```
SLACK_SUCCESS_CHANNEL
SLACK_FAILURE_CHANNEL
```

Reply:
Any opinions on whether to use alist-staging vs alist-production here? Also relevant, for SLACK_CHANNEL, #staging-releases or ...?